### PR TITLE
Add cgroupsPath validation

### PR DESCRIPTION
**Pull request by q384566678**
Implement the following specerror:

CgroupsAbsPathRelToMount: In the case of an absolute path (starting with `/`), the runtime MUST take the path to be relative to the cgroups mount point.

 CgroupsPathAttach: If the value is specified, the runtime MUST consistently attach to the same place in the cgroups hierarchy given the same value of `cgroupsPath`.

Signed-off-by: Zhou Hao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

**Comment by q384566678** [comment]
@liangchenye PTAL

File: cgroups/cgroups_v1.go [hunk 1]
```diff
@@ -193,6 +207,14 @@ func (cg *CgroupV1) GetBlockIOData(pid int, cgPath string) (*rspec.LinuxBlockIO,
 
 // GetCPUData gets cgroup cpus data
 func (cg *CgroupV1) GetCPUData(pid int, cgPath string) (*rspec.LinuxCPU, error) {
+	if filepath.IsAbs(cgPath) {
+		path := filepath.Join(cg.MountPath, "cpu", cgPath)
+		if _, err := os.Stat(path); err != nil {
+			if os.IsNotExist(err) {
+				return nil, specerror.NewError(specerror.CgroupsAbsPathRelToMount, fmt.Errorf("In the case of an absolute path, the runtime MUST take the path to be relative to the cgroups mount point"), rspec.Version)
+			}
```
Comment (liangchenye) [inline cgroups/cgroups_v1.go]: same, return error in case of other error.

File: cgroups/cgroups_v1.go [hunk 2]
```diff
@@ -349,6 +375,14 @@ func getHugePageSize() ([]string, error) {
 
 // GetHugepageLimitData gets cgroup hugetlb data
 func (cg *CgroupV1) GetHugepageLimitData(pid int, cgPath string) ([]rspec.LinuxHugepageLimit, error) {
+	if filepath.IsAbs(cgPath) {
+		path := filepath.Join(cg.MountPath, "hugetlb", cgPath)
+		if _, err := os.Stat(path); err != nil {
+			if os.IsNotExist(err) {
+				return nil, specerror.NewError(specerror.CgroupsAbsPathRelToMount, fmt.Errorf("In the case of an absolute path, the runtime MUST take the path to be relative to the cgroups mount point"), rspec.Version)
+			}
```
Comment (liangchenye) [inline cgroups/cgroups_v1.go]: return error in case of other error

File: cgroups/cgroups_v1.go [hunk 3]
```diff
@@ -386,6 +424,14 @@ func (cg *CgroupV1) GetHugepageLimitData(pid int, cgPath string) ([]rspec.LinuxH
 
 // GetMemoryData gets cgroup memory data
 func (cg *CgroupV1) GetMemoryData(pid int, cgPath string) (*rspec.LinuxMemory, error) {
+	if filepath.IsAbs(cgPath) {
+		path := filepath.Join(cg.MountPath, "memory", cgPath)
+		if _, err := os.Stat(path); err != nil {
+			if os.IsNotExist(err) {
+				return nil, specerror.NewError(specerror.CgroupsAbsPathRelToMount, fmt.Errorf("In the case of an absolute path, the runtime MUST take the path to be relative to the cgroups mount point"), rspec.Version)
+			}
```
Comment (liangchenye) [inline cgroups/cgroups_v1.go]: same here

File: cgroups/cgroups_v1.go [hunk 4]
```diff
@@ -468,6 +518,14 @@ func (cg *CgroupV1) GetMemoryData(pid int, cgPath string) (*rspec.LinuxMemory, e
 
 // GetNetworkData gets cgroup network data
 func (cg *CgroupV1) GetNetworkData(pid int, cgPath string) (*rspec.LinuxNetwork, error) {
+	if filepath.IsAbs(cgPath) {
+		path := filepath.Join(cg.MountPath, "net_cls", cgPath)
+		if _, err := os.Stat(path); err != nil {
+			if os.IsNotExist(err) {
+				return nil, specerror.NewError(specerror.CgroupsAbsPathRelToMount, fmt.Errorf("In the case of an absolute path, the runtime MUST take the path to be relative to the cgroups mount point"), rspec.Version)
+			}
```
Comment (liangchenye) [inline cgroups/cgroups_v1.go]: and here

File: cgroups/cgroups_v1.go [hunk 5]
```diff
@@ -526,6 +588,14 @@ func (cg *CgroupV1) GetNetworkData(pid int, cgPath string) (*rspec.LinuxNetwork,
 
 // GetPidsData gets cgroup pids data
 func (cg *CgroupV1) GetPidsData(pid int, cgPath string) (*rspec.LinuxPids, error) {
+	if filepath.IsAbs(cgPath) {
+		path := filepath.Join(cg.MountPath, "pids", cgPath)
+		if _, err := os.Stat(path); err != nil {
+			if os.IsNotExist(err) {
+				return nil, specerror.NewError(specerror.CgroupsAbsPathRelToMount, fmt.Errorf("In the case of an absolute path, the runtime MUST take the path to be relative to the cgroups mount point"), rspec.Version)
+			}
```
Comment (liangchenye) [inline cgroups/cgroups_v1.go]: the last place.
Besides these, it looks good to me.

Comment (q384566678) [reply cgroups/cgroups_v1.go]: updated, PTAL.

**Comment by liangchenye** [comment]
LGTM

**Pull request merged**